src/main/java/validation/IndentationValidity.java

```diff
@@ -1,28 +1,217 @@
 package validation;
 
+import java.util.Stack;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 /**
  * Check if the indentation is valid or not.
  */
+@SuppressWarnings("PMD.GuardLogStatement")
 public final class IndentationValidity implements Validation {
-
+	
 	public IndentationValidity(final String code) {
 		this.code = code;
 	}
 	
 	@Override
 	public boolean valid() {
-		throw new UnsupportedOperationException("Not implemented yet.");
+		this.lines = code.split("\n");   	// code split
+		this.lineLength = this.lines.length;	
+		
+		calcIndentUnit();			// Calculate Base Indent
+		
+		if(firstLineHaveIndent()) {				// first line doesn't have indent
+			setError("First line should not have indentation", 1);
+			this.validity = false;
+			return false;
+		}
+		final Stack<Integer> stack = new Stack<Integer>();
+		
+		for (int currentLine=0;currentLine<this.lineLength;currentLine++) {	
+			
+			final IndentAmount amount = this.indentAmount(lines[currentLine]);	
+			
+			if(!amount.valid()) {					
+				setError("Indentation Type Miss Match", currentLine+1);
+				this.validity = false;
+				return false;
+			} 
+			
+			final int currentIndent = amount.value();
+			if(stack.isEmpty()) {
+				stack.addElement(currentIndent);
+			}else{
+				while(currentIndent<stack.lastElement()) {
+					stack.pop();
+					
+				}
+				
+				if(currentIndent>stack.lastElement()) {
+					if(currentIndent - this.baseIndent != stack.lastElement()) {
+						setError("Please check the indentation.", currentLine+1);
+						this.validity =false;
+						return false;
+					}
+						
+					stack.addElement(currentIndent);
+					
+				}
+				
+			}
+				
+			
+		}
+		return this.validity;
 	}
-
 	@Override
 	public int line() {
-		throw new UnsupportedOperationException("Not implemented yet.");
+		if(this.validity) {
+			throw new IllegalStateException("The indentation should be valid.");
+		}else {
+			return this.errorLine;			
+		}
 	}
-
 	@Override
 	public String reason() {
-		throw new UnsupportedOperationException("Not implemented yet.");
+		if(this.validity) {
+			throw new IllegalStateException("The indentation should be valid.");
```

Comment:
The message is wrong.
How about "This method can be used when the indentation is invalid."?